dev/gendoc.py

```diff
@@ -0,0 +1,195 @@
+            release_doc = pypandoc.convert_text(body, "rst", format="md")
+
+            # Make PR reference link pretty.
+            # Replace ", #..." to ", `...<https://github.com/databricks/koalas/pull/...>`_"
+            release_doc = re.sub(
+                r', #(\d+)',
+                r', `#\1 <https://github.com/databricks/koalas/pull/\1>`_', release_doc)
+            # Replace "(#..." to "(`...<https://github.com/databricks/koalas/pull/...>`_"
+            release_doc = re.sub(
+                r'\(#(\d+)',
+                r'(`#\1 <https://github.com/databricks/koalas/pull/\1>`_', release_doc)
+
+            index_file.write("   " + tag_name)
+            index_file.write("\n")
+            index_file.write("\n")
+
+            with open("%s/%s.rst" % (whatsnew_dir, tag_name), "a") as release_file:
+                release_file.write("=" * len(name))
+                release_file.write("\n")
+                release_file.write(name)
+                release_file.write("\n")
+                release_file.write("=" * len(name))
+                release_file.write("\n")
+                release_file.write("\n")
+                release_file.write(release_doc)
+                release_file.write("\n")
+                release_file.write("\n")
+
+
+def download_pandoc_if_needed(path):
+    """
+    Download pandoc that is used by pypandoc.
+    """
+
+    # This is to patch https://github.com/bebraw/pypandoc/pull/154.
+    # We can remove when pypandoc is upgraded.
+    def _handle_linux(filename, targetfolder):
+
+        print("* Unpacking %s to tempfolder..." % (filename))
+
+        tempfolder = tempfile.mkdtemp()
+        cur_wd = os.getcwd()
+        filename = os.path.abspath(filename)
+        try:
+            os.chdir(tempfolder)
+            cmd = ["ar", "x", filename]
+            # if only 3.5 is supported, should be `run(..., check=True)`
+            subprocess.check_call(cmd)
+            files = os.listdir(".")
+            archive_name = next(x for x in files if x.startswith('data.tar'))
+            cmd = ["tar", "xf", archive_name]
+            subprocess.check_call(cmd)
+            # pandoc and pandoc-citeproc are in ./usr/bin subfolder
+            for exe in ["pandoc", "pandoc-citeproc"]:
+                src = os.path.join(tempfolder, "usr", "bin", exe)
+                dst = os.path.join(targetfolder, exe)
+                print("* Copying %s to %s ..." % (exe, targetfolder))
+                shutil.copyfile(src, dst)
+                pandoc_download._make_executable(dst)
+            src = os.path.join(tempfolder, "usr", "share", "doc", "pandoc", "copyright")
+            dst = os.path.join(targetfolder, "copyright.pandoc")
+            print("* Copying copyright to %s ..." % (targetfolder))
+            shutil.copyfile(src, dst)
+        finally:
+            os.chdir(cur_wd)
+            shutil.rmtree(tempfolder)
+
+    pandoc_urls, _ = pandoc_download._get_pandoc_urls("latest")
```

Comment:
This logic was just ported from pypandoc to:

1. check if already downloaded
2. to patch `https://github.com/bebraw/pypandoc/pull/154`